motionplan/tpSpaceRRT.go

```diff
@@ -172,185 +205,329 @@ func (mp *tpSpaceRRTMotionPlanner) planRunner(
 		}
 	}
 
-	dist := math.Sqrt(mp.planOpts.DistanceFunc(&Segment{StartPosition: startPose, EndPosition: goalPose}))
-	midPt := goalPose.Point().Sub(startPose.Point())
+	m1chan := make(chan *nodeAndError, 1)
+	m2chan := make(chan *nodeAndError, 1)
+	defer close(m1chan)
+	defer close(m2chan)
+
+	dist := math.Sqrt(mp.planOpts.DistanceFunc(&ik.Segment{StartPosition: startPose, EndPosition: goalPose}))
```

Comment:
Shouldn't the distance function either be responsible for taking the sqrt if necessary?

Reply:
Generally yes. In this case we're just defining a region within which to sample, and it's better to under-estimate than over estimate. This is the only place we sqrt the distance, and it's deliberate here.